SimPEG/potential_fields/gravity/simulation.py

```diff
@@ -27,27 +76,134 @@ class Simulation3DIntegral(BasePFSimulation):
+        to ``"disk"``. Default to "./sensitivities".
+    engine : str, optional
+       Choose which engine should be used to run the forward model:
+       ``"geoana"`` or "``choclo``".
+    choclo_parallel : bool, optional
+        If True, the simulation will run in parallel. If False, it will
+        run in serial. If ``engine`` is not ``"choclo"`` this argument will be
+        ignored.
     """
 
     rho, rhoMap, rhoDeriv = props.Invertible("Density")
 
-    def __init__(self, mesh, rho=None, rhoMap=None, **kwargs):
+    def __init__(
+        self,
+        mesh,
+        rho=None,
+        rhoMap=None,
+        engine="geoana",
+        choclo_parallel=True,
+        **kwargs,
+    ):
         super().__init__(mesh, **kwargs)
         self.rho = rho
         self.rhoMap = rhoMap
         self._G = None
         self._gtg_diagonal = None
         self.modelMap = self.rhoMap
+        self.choclo_parallel = choclo_parallel
+        self.engine = engine
+        self._sanity_checks_engine(kwargs)
+        # Define jit functions
+        if self.engine == "choclo":
+            if choclo_parallel:
+                self._sensitivity_gravity = _sensitivity_gravity_parallel
+                self._forward_gravity = _forward_gravity_parallel
+            else:
+                self._sensitivity_gravity = _sensitivity_gravity_serial
+                self._forward_gravity = _forward_gravity_serial
+
+    def _sanity_checks_engine(self, kwargs):
+        """
+        Sanity checks for the engine parameter.
+
+        Needs the kwargs passed to the __init__ method to raise some warnings.
+        Will set n_processes to None if it's present in kwargs.
+        """
+        if self.engine not in ("choclo", "geoana"):
+            raise ValueError(
+                f"Invalid engine '{self.engine}'. Choose from 'geoana' or 'choclo'."
+            )
+        if self.engine == "choclo" and choclo is None:
+            raise ImportError(
+                "The choclo package couldn't be found."
+                "Running a gravity simulation with 'engine=\"choclo\"' needs "
+                "choclo to be installed."
+                "\nTry installing choclo with:"
+                "\n    pip install choclo"
+                "\nor:"
+                "\n    conda install choclo"
+            )
+        # Warn if n_processes has been passed
+        if self.engine == "choclo" and "n_processes" in kwargs:
+            warnings.warn(
+                "The 'n_processes' will be ignored when selecting 'choclo' as the "
+                "engine in the gravity simulation.",
+                UserWarning,
+                stacklevel=1,
+            )
+            self.n_processes = None
+        # Sanity checks for sensitivity_path when using choclo and storing in disk
+        if self.engine == "choclo" and self.store_sensitivities == "disk":
+            if os.path.isdir(self.sensitivity_path):
```

Comment:
Feels like the same check should happen regardless if the engine is choclo or geoana. A bit of change of behaviour, but generally safer. I like it.

Reply:
I'll keep that in mind! We should explore adding this to the parent class then.